client/client_inventory.go

```diff
@@ -60,16 +77,43 @@ func doSubmitInventory(
 	api ApiRequester,
 	method, url string,
 	data interface{},
+	retryInterval time.Duration,
+	tries int,
 ) (*http.Response, error) {
+	const defaultTries = 3
 	req, err := makeInventorySubmitRequest(method, url, data)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to prepare inventory submit request")
 	}
 
 	r, err := api.Do(req)
+	outputTries := tries
+	if outputTries == 0 {
+		outputTries = defaultTries //this is the default from the GetExponentialBackoffTime
+	}
```

Comment:
That's not how it works. It tries three times at each interval, including three times at the max interval. If you really want to print this information, make a function next to `GetExponentialBackoffTime`:

```golang
func GetExponentialBackoffCount(maxInterval, maxRetries int) int {
    if maxRetries > 0 {
        return maxRetries
    }
    if maxInterval < 1 * time.Minute {
        maxInterval = 1 * time.Minute
    }
    return ceil(log2(maxInterval / time.Minute) + 1) * 3
}
```

I'm not exactly sure what those math functions are called in Go, but you get my point.

OTOH, you could also just omit this information.

Reply:
I will omit then.